sdcm/nemesis.py

```diff
@@ -5147,6 +5151,53 @@ def disrupt_disable_binary_gossip_execute_major_compaction(self):
             self.target_node.restart_scylla_server()
             raise
 
+    def disrupt_grow_shrink_znodes(self):
+        """"Add/remove znodes to same dc where target node"""
+        duration_with_znode = 300
+        new_znode = self._add_and_init_new_cluster_nodes(count=1, zero_token=True)[0]
+        self.log.debug("Run with zero-token node %s for %ds", new_znode.name, duration_with_znode)
+        znode = random.choice([node for node in self.cluster.zero_nodes if node.dc_idx == self.target_node.dc_idx])
+        self.decommission_nodes(nodes=[znode])
+
+    def disrupt_add_remove_dc_with_znode(self) -> None:
+        if self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("Operator doesn't support multi-DC yet. Skipping.")
+        if self.cluster.test_config.MULTI_REGION:
+            raise UnsupportedNemesis(
+                "add_remove_dc skipped for multi-dc scenario (https://github.com/scylladb/scylla-cluster-tests/issues/5369)")
+        InfoEvent(message='Starting New DC Nemesis').publish()
+        node = self.cluster.data_nodes[0]
+        system_keyspaces = ["system_distributed", "system_traces"]
+        if not node.raft.is_consistent_topology_changes_enabled:  # auth-v2 is used when consistent topology is enabled
+            system_keyspaces.insert(0, "system_auth")
+        self._switch_to_network_replication_strategy(self.cluster.get_test_keyspaces() + system_keyspaces)
+        datacenters = list(self.tester.db_cluster.get_cluster_status().keys())
+        node_added = False
+        with ExitStack() as context_manager:
+            def finalizer(exc_type, *_):
+                # in case of test end/killed, leave the cleanup alone
+                if exc_type is not KillNemesis:
+                    # with self.cluster.cql_connection_patient(node) as session:
+                    #     session.execute('DROP KEYSPACE IF EXISTS keyspace_new_dc')
+                    if node_added:
+                        self.cluster.decommission(new_node)
+            context_manager.push(finalizer)
+
+            new_node = self._add_new_node_in_new_dc(zero_token=True)
+            node_added = True
+            status = self.tester.db_cluster.get_cluster_status()
+            new_dc_list = [dc for dc in list(status.keys()) if dc.endswith("_nemesis_dc")]
+            assert new_dc_list, "new datacenter was not registered"
```

Comment:
is this the only thing to assert here? shouldn't we verify this node got some raft status or something?

Reply:
removed